daemon/daemonbuilder/builder.go

```diff
@@ -210,10 +210,10 @@ func (d Docker) GetCachedImage(imgID string, cfg *runconfig.Config) (string, err
 // DetectContextFromRemoteURL returns a context and in certain cases the name of the dockerfile to be used
 // irrespective of user input.
 // progressReader is only used if remoteURL is actually a URL (not empty, and not a Git endpoint).
-func DetectContextFromRemoteURL(r io.ReadCloser, remoteURL string, progressReader *progressreader.Config) (context builder.ModifiableContext, dockerfileName string, err error) {
+func DetectContextFromRemoteURL(reader io.ReadCloser, writer io.Writer, remoteURL string, size int64) (context builder.ModifiableContext, dockerfileName string, err error) {
```

Comment:
previous and current naming of r/w is bad. There is literally no clue what they really are, nor in docstring, neither in their names. So, if we want to impove naming probably we should call it somehow related to their meaning or just leave it r/w.